app/services/blacklight/search_service.rb

```diff
@@ -0,0 +1,120 @@
+# frozen_string_literal: true
+# SearchService returns search results from the repository
+module Blacklight
+  class SearchService
+    include Blacklight::RequestBuilders
+
+    def initialize(blacklight_config)
+      @blacklight_config = blacklight_config
+    end
+
+    # TODO: Can this be private?
+    attr_reader :blacklight_config
+
+    # a solr query method
+    # @param [Hash] user_params ({}) the user provided parameters (e.g. query, facets, sort, etc)
+    # @yield [search_builder] optional block yields configured SearchBuilder, caller can modify or create new SearchBuilder to be used. Block should return SearchBuilder to be used.
+    # @return [Blacklight::Solr::Response] the solr response object
+    def search_results(user_params)
```

Comment:
Do you think it would make sense to add these `user_params` (also called `request_params`, or `paging_params`?) as an argument to the initializer?